### PR TITLE
Move Toast State to Store

### PR DESCRIPTION
**Pull request by jhbertra**
- Adds the current toast message to the Store
- Makes Toast its own component that subscribes to the store.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Comment by palas** [comment]
This PR does have an error to fix before we can merge: https://github.com/input-output-hk/marlowe-cardano/runs/4778469661?check_suite_focus=true

File: marlowe-dashboard-client/src/Toast/State.purs [hunk 1]
```diff
@@ -39,18 +60,21 @@ toastTimeoutSubscription toast =
...
+  => MonadStore Store.Action Store.Store m
   => Action
   -> HalogenM State Action slots msg m Unit
-handleAction (AddToast toast) = do
-  timeoutSubscription <- subscribe $ toastTimeoutSubscription toast
-  assign _mToast (Just { message: toast, expanded: false, timeoutSubscription })
+handleAction (Receive mToast) = do
+  H.put $ deriveState mToast
+  sub <- traverse (subscribe <<< toastTimeoutSubscription) mToast
+  H.modify_ _ { timeoutSubscription = sub }
```
Comment (palas) [inline marlowe-dashboard-client/src/Toast/State.purs:69]: Is it a danger that we overwrite a previous subscription here? Do we need to check and unsubscribe or it cannot happen?

**Review by palas — APPROVED** [review]
Looks good, just a question to be sure

**Review by bwbush — APPROVED** [review]
This looks good to me, but I don't have much relevant expertise.

**Pull request merged**